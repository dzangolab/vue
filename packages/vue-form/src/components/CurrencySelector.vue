<template>
  <div class="currency-selector">
    <SelectInput
      v-bind="selectionOptions"
      :disabled="disabled"
      :label="label"
      :model-value="modelValue"
      :multiple="multiple"
      :name="name"
      :options="options"
      :placeholder="placeholder"
      :schema="schema"
      @update:model-value="$emit('update:modelValue', $event)"
    >
      <template
        v-for="(option, index) in options"
        :key="`${index}-${option.label}`"
        #[option.value]
      >
        <div class="currency-label">
          <span
            v-if="option?.[optionLabelOrder[0]]"
            :class="[optionLabelOrder[0], 'primary-currency-label']"
          >
            {{ option?.[optionLabelOrder[0]] }}
          </span>
          <span
            v-if="option?.[optionLabelOrder[1]]"
            :class="[optionLabelOrder[1], 'secondary-currency-label']"
          >
            {{ option?.[optionLabelOrder[1]] }}
          </span>
          <span
            v-if="option?.[optionLabelOrder[2]]"
            :class="[optionLabelOrder[2], 'tertiary-currency-label']"
          >
            {{ option?.[optionLabelOrder[2]] }}
          </span>
        </div>
      </template>
    </SelectInput>
  </div>
</template>

<script lang="ts">
export default {
  name: "CurrencySelector",
};
</script>

<script setup lang="ts">
import { z } from "zod";

import SelectInput from "./SelectInput.vue";

import type { CurrencyOption } from "../types";
import type { PropType } from "vue";

defineProps({
  disabled: {
    default: false,
    type: Boolean,
  },
  label: {
    default: "",
    required: false,
    type: String as PropType<string>,
  },
  modelValue: {
    default: () => null,
    required: false,
    type: [Boolean, Number, String, Array] as PropType<
      boolean | number | string | (number | string)[]
    >,
  },
  multiple: {
    default: false,
    type: Boolean,
  },
  name: {
    default: "currency",
    required: false,
    type: String as PropType<string>,
  },
  optionLabelOrder: {
    type: Array as PropType<Array<keyof CurrencyOption>>,
    default: () => ["shortName", "label", "symbol"],
  },
  options: {
    required: true,
    type: Array as PropType<CurrencyOption[]>,
  },
  placeholder: {
    default: undefined,
    type: String,
  },
  schema: {
    default: () => {
      return {};
    },
    required: false,
    type: Object as PropType<z.ZodType<string | number | string[] | number[]>>,
  },
  selectionOptions: {
    default: () => ({}),
    type: Object,
  },
});

defineEmits(["update:modelValue"]);
</script>

<style lang="css">
@import "../assets/css/currencySelector.css";
</style>
