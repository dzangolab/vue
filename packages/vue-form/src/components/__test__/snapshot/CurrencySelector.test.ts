import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import CurrencySelector from "../../CurrencySelector.vue";

import type { CurrencyOption } from "../../../types";

describe("SelectInput", () => {
    const options = [
        { code: "AUD", label: "Australian Dollar", symbol: "$", value: "AUD" },
        { code: "GBP", label: "British Pound", symbol: "£", value: "GBP" },
        { code: "EUR", disabled: true, label: "Euro", symbol: "€", value: "EUR" },
        { code: "JPY", label: "Japanese Yen", symbol: "¥", value: "JPY" },
        { code: "USD", label: "US Dollar", symbol: "$", value: "USD" },
    ] as CurrencyOption[];

    it("matches snapshot", async () => {
        const wrapper = mount(CurrencySelector, {
            props: {
                label: "Currency Selector",
                modelValue: ["AUD", "GBP"],
                multiple: true,
                name: "currency-selector",
                options: options,
                placeholder: "Select a currency",
            },
        });

    expect(wrapper.element).toMatchSnapshot();
  });
});
