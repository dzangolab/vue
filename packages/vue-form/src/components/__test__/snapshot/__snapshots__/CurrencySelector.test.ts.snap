// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectInput > matches snapshot 1`] = `
<div
  class="currency-selector"
>
  <div
    class="field currency-selector"
  >
    <label
      for="currency-selector"
    >
      Currency Selector
    </label>
    
    <div
      class="multiple-mode multiselect"
      id="input-field-currency-selector"
      name="currency-selector"
      tabindex="0"
      value="AUD,GBP"
    >
      <!--v-if-->
      <div
        class="multiselect-input"
      >
        <span
          class="multiselect-placeholder"
        >
          Select a currency
        </span>
        <span
          class="action-items"
        >
          <!--v-if-->
          <svg
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Complete"
            >
              <g
                id="F-Chevron"
              >
                <polyline
                  fill="none"
                  id="down"
                  points="5 8.5 12 15.5 19 8.5"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </g>
            </g>
          </svg>
        </span>
      </div>
      <!--v-if-->
    </div>
    <!---->
    
  </div>
</div>
`;
